src/File.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools;
+
+class File
+{
+    /**
+     * Reads file resource line by line.
+     *
+     * @param resource $fileResource
+     * @param int<0, max>|null $length
+     *
+     * @return \Generator<string>
+     *
+     * @see fgets()
+     */
+    public static function readByLine($fileResource, ?int $length = null): \Generator
+    {
+        // @phpstan-ignore-next-line (expects int<0, max>, int<0, max>|null given.)
+        while (($line = @\fgets($fileResource, $length)) !== false) {
+            yield $line;
+        }
+    }
+
+    /**
+     * Reads data from CSV file resource like fgetcsv() function.
+     *
+     * @param resource $fileResource
+     * @param int<0, max>|null $length
+     * @param string $separator
+     * @param string $enclosure
+     * @param string $escape
+     *
+     * @return \Generator<array<int, string|null>>
+     *
+     * @throws \RuntimeException if read error occurred
+     *
+     * @see fgetcsv()
+     */
+    public static function readCsv(
+        $fileResource,
+        ?int $length = null,
+        string $separator = ",",
+        string $enclosure = "\"",
+        string $escape = "\\"
```

Comment:
What do you think about rearranging the parameters to put the more-likely-to-be-used parameters first:
```php
readCsv(resource $file, string $separator = ',', string $enclosure = '"', string $escape = '\') { ... }
```
And how about removing length?

Reply:
Agree. Done.